### PR TITLE
Re-add support for fish shell

### PR DESCRIPTION
**Pull request by iamed2**
It was removed in https://github.com/JuliaLang/julia/pull/4924

This adds it back using the `begin; ls; true; end` pattern instead of the `(ls) && true` pattern.

### CODE REVIEW

File: base/client.jl [hunk 1]
```diff
@@ -120,11 +114,19 @@ function repl_cmd(cmd, out)
...
     else
-        run(ignorestatus(@static is_windows() ? cmd : (isa(STDIN, TTY) ? `$shell -i -c "($(shell_escape(cmd))) && true"` : `$shell -c "($(shell_escape(cmd))) && true"`)))
+        run(ignorestatus(@static is_windows() ? cmd : (isa(STDIN, TTY) ? `$shell -i -c "$(shell_wrap_true(shell_name, cmd))"` : `$shell -c "$(shell_wrap_true(shell_name, cmd))"`)))
     end
     nothing
 end
 
+function shell_wrap_true(shell_name, cmd)
+    if shell_name == "fish"
+        "begin; $(shell_escape(cmd)); true; end"
```
Comment (vchuravy) [inline base/client.jl]: wouldn't it need to be `begin; $(shell_escape(cmd)); and true; end` to be the same in both cases?

Comment (iamed2) [inline base/client.jl]: Yep

**Comment by omus** [comment]
Would be nice to have some tests

**Review by StefanKarpinski — APPROVED** [review]

**Comment by vchuravy** [comment]
This will go well with https://github.com/JuliaLang/julia/pull/19786#issuecomment-270119383, has anybody run the whole testsuite under fish?

**Comment by iamed2** [comment]
Tests fail. Trying to figure out why. Seems like it's only happening when the test suite is run, not individual components, so might be something in the parallel test-running code.

**Comment by tkelman** [comment]
Are we sure we want to enable this if the tests still fail?

**Comment by iamed2** [comment]
I was trying to run tests but they were breaking. That issue was fixed, but I didn't get a change to rerun before I went on holiday. There could very well still be a breakage.

**Comment by StefanKarpinski** [comment]
The tests were green so I assumed they had been fixed since that comment. It would be great if people could use reviews to indicate that changes are required on PRs, especially ones with green tests.

**Comment by tkelman** [comment]
Our CI isn't using fish as a shell

> run the whole testsuite under fish?

**Comment by iamed2** [comment]
Mentioned in https://github.com/JuliaLang/julia/issues/4918 that I ran `Base.runtests()` using fish today and it succeeded.

**Pull request merged**